File: hw1/loans.py
import pandas as pd

loans=pd.read_csv('data/loans.csv')

tidy_loans=pd.melt(loans,id_vars=['id','account_id','date','amount','payments'])
tidy_loans=tidy_loans.loc[tidy_loans.loc[:,'value']!='-',]
temp=tidy_loans.variable.str.split('_')
month=temp.str.get(0)
status=temp.str.get(1)
tidy_loans['month']=month
tidy_loans['status']=status
tidy_loans=tidy_loans.drop(columns=['variable','value'])
tidy_loans.to_csv('loans_py.csv',index=False)

File: hw1/loans.R
library(rstudioapi)
setwd(dirname(rstudioapi::getActiveDocumentContext()$path))

library(reshape)
library(dplyr)
library(stringr)
loans<-read.csv('data/loans.csv')

colnames(loans)

head(loans)

tidy_loans <- melt(loans,id=c('id','account_id','date','amount',
                              'payments'))

tidy_loans <- filter(tidy_loans,value !='-')
variable<-str_split_fixed(as.character(tidy_loans$variable),'_',2)

month=variable[,1]
month=str_sub(month,-2)
status=variable[,2]

tidy_loans=select(tidy_loans,-c(variable,value))
tidy_loans['month']=month
tidy_loans['status']=status

write.csv(tidy_loans,'loans_r.csv',row.names=FALSE)
